core/includes/form.inc

```diff
@@ -3001,7 +3001,7 @@ function theme_radio($variables) {
   $element['#attributes']['type'] = 'radio';
   element_set_attributes($element, array('id', 'name', '#return_value' => 'value'));
 
-  if (isset($element['#return_value']) && $element['#value'] !== FALSE && $element['#value'] == $element['#return_value']) {
+  if (isset($element['#return_value']) && $element['#value'] !== FALSE && !strcmp($element['#value'], $element['#return_value'])) {
```

Comment:
Can we please add an inline comment here, explaining the issue and why exactly we are using `strcmp` instead of a simple comparison operator?

Reply:
Comment added.

TBH, the documentation page for `#options` property should warn people off about mixing integer and string keys, because even apart from this one, someone, somewhere is going to try to compare two key values for some reason and encounter this peculiarity.

Reply:
Warning has been added to the documentation page, so marking this as resolved.